ovs/lib/storagedriver.py

```diff
@@ -399,6 +408,200 @@ def _configure_arakoon_to_volumedriver(cluster_name):
+            storagedriver.storagedriver_id = vrouter_id
+            storagedriver.save()
+
+            # ALBA Proxies
+            proxy_service_type = ServiceTypeList.get_by_name(ServiceType.SERVICE_TYPES.ALBA_PROXY)
+            for proxy_id in xrange(amount_of_proxies):
+                service = Service()
+                service.storagerouter = storagerouter
+                service.ports = [ports[4 + proxy_id]]
+                service.name = 'albaproxy_{0}_{1}'.format(vpool.name, proxy_id)
+                service.type = proxy_service_type
+                service.save()
+                alba_proxy = AlbaProxy()
+                alba_proxy.service = service
+                alba_proxy.storagedriver = storagedriver
+                alba_proxy.save()
+
+        return storagedriver
+
+    @staticmethod
+    def configure_storagedriver_partitions(storagedriver_guid, fragment_cache_settings, block_cache_settings, mountpoint_settings,
+                                           amount_of_proxies, partition_info=None):
+        """
+        Configure all partitions for a storagedriver
+        Attempts to clean up in case of errors
+        :param storagedriver_guid: Guid of the Storagedriver
+        :type storagedriver_guid: str
+        :param fragment_cache_settings: Settings of the fragment cache ({is_backend: bool, read: bool, write: bool})
+        :type fragment_cache_settings: dict
+        :param block_cache_settings: Settings of the block cache ({is_backend: bool, read: bool, write: bool})
+        :type block_cache_settings: dict
+        :param mountpoint_settings: Settings about the cache ({mountpoint_cache: , writecache_size_requested: , largest_write_mountpoint}
+        :type mountpoint_settings: dict
+        :param amount_of_proxies: Amount of proxies to deploy
+        :type amount_of_proxies: int
+        :param partition_info: Information about the partitions (Optional, won't query for the info if supplied)
+        :type partition_info: dict
+        :raises: ValueError: - When calculating the cache sizes went wrong
+        :return: Dict with information about the created items
+        :rtype: dict
+        """
+        # * Information about backoff_gap and trigger_gap (Reason for 'smallest_write_partition' introduction)
+        # * Once the free space on a mount point is < trigger_gap (default 1GiB), it will be cleaned up and the cleaner attempts to
+        # * make sure that <backoff_gap> free space is available => backoff_gap must be <= size of the partition
+        # * Both backoff_gap and trigger_gap apply to each mount point individually, but cannot be configured on a per mount point base
+        # Assign WRITE / Fragment cache
+        from ovs.lib.storagerouter import StorageRouterController  # Avoid circular reference
+        storagedriver = StorageDriver(storagedriver_guid)
+        storagerouter = storagedriver.storagerouter
+        if partition_info is None:
```

Comment:
There's only 1 usage of this function in which you always pass the partition information